crates/bevy_math/src/curve/mod.rs

```diff
@@ -968,6 +968,73 @@ where
     }
 }
 
+/// Convert a `start` and `end` point of some type into a [`Curve`], sampled by interpolating
+/// between the two boundary values and under the application of an easing function.
+pub fn easing_curve<T: Clone>(
+    domain: Interval,
+    start: T,
+    end: T,
+    easing_fn: impl Fn(f32) -> f32,
+) -> impl Curve<T>
+where
+    T: std::ops::Mul<f32, Output = T>
+        + std::ops::Add<T, Output = T>
+        + std::ops::Div<f32, Output = T>,
+{
+    let interpolate_with_ease = move |t: f32| {
+        let len = domain.length();
+        let start_factor = (domain.end() - t) / len;
+        let end_factor = (t - domain.start()) / len;
+        start.clone() * easing_fn(start_factor) + end.clone() * easing_fn(end_factor)
+    };
+    FunctionCurve {
+        domain,
+        f: interpolate_with_ease,
+        _phantom: PhantomData,
+    }
+}
+
+/// Convert a `start` and `end` point of some type into a [`Curve`], sampled by linearly
+/// interpolating between the two boundary values.
+pub fn line_curve<T: Clone>(domain: Interval, start: T, end: T) -> impl Curve<T>
+where
+    T: std::ops::Mul<f32, Output = T>
+        + std::ops::Add<T, Output = T>
+        + std::ops::Div<f32, Output = T>,
+{
+    easing_curve(domain, start, end, easing_functions::linear_ease)
+}
+
+/// A simple collection of easing functions. Inspiration taken from [this website]
+///
+/// [this website]: https://notes.yvt.jp/Graphics/Easing-Functions/
+pub mod easing_functions {
```

Comment:
I feel like it might be more natural to make the easing functions themselves into curves and relate them to the linear curve via `reparametrize_by_curve` (since that's essentially what it's for) instead of this organization where the "basic object" is the easing function.

That is, my suggestion would be to have `linear_curve` be the basic thing, then define what is currently `easing_curve` using `linear_curve(...).reparametrize_by_curve`.

Reply:
That's actually really clean as well although the easing curves are a bit repetitive code-wise. I'll change it!